### PR TITLE
Update npm dependencies

### PR DESCRIPTION
**Pull request by DivineDominion**
Addresses all low hanging fruit for #2095, bumping versions that work well together.

I ran a couple of clean `npm install` and also `npm update` and `npm audit` changes.

Node v17+ also wouldn't allow `react-scripts build` etc. and an apparent fix is to disable the security check:
https://stackoverflow.com/questions/69692842/error-message-error0308010cdigital-envelope-routinesunsupported

I did that in the package scripts section, too. Thing is: `react-scripts` probably need to go anyway because there's effectively no maintenance and dependency updates have been unaddressed for years. react.dev recomment Vite nowadays instead of "Create React App", so supposedly, Vite is going to be the future? https://cathalmacdonnacha.com/migrating-from-create-react-app-cra-to-vite -- but that's not for today :)

Oh, `npm run build` and `npm run start` work fine 👍

### CODE REVIEW

File: src/webapp/package.json [hunk 1]
```diff
@@ -5,30 +5,30 @@
...
   },
   "scripts": {
-    "start": "react-scripts start",
-    "build": "react-scripts build",
-    "test": "react-scripts test",
-    "eject": "react-scripts eject"
+    "start": "react-scripts --openssl-legacy-provider start",
+    "build": "react-scripts --openssl-legacy-provider build",
+    "test": "react-scripts --openssl-legacy-provider test",
+    "eject": "react-scripts --openssl-legacy-provider eject"
```
[the remark this answers is not in the document]
Comment (DivineDominion) [reply src/webapp/package.json:31]: Node v17+ also wouldn't allow react-scripts build etc. and an apparent fix is to disable the security check:
https://stackoverflow.com/questions/69692842/error-message-error0308010cdigital-envelope-routinesunsupported

Without this, it just crashes. 

Alternative: downgrade Node.

File: src/webapp/package.json [hunk 2]
```diff
@@ -5,30 +5,30 @@
   "dependencies": {
     "@emotion/react": "^11.10.6",
     "@emotion/styled": "^11.10.6",
-    "@mui/icons-material": "^5.11.16",
-    "@mui/material": "^5.12.0",
-    "@mui/styles": "^5.12.0",
-    "@testing-library/jest-dom": "^5.16.5",
-    "@testing-library/react": "^12.1.2",
+    "@mui/icons-material": "^5.14.16",
+    "@mui/material": "^5.14.17",
```
[the remark this answers is not in the document]
Comment (pabera) [reply src/webapp/package.json:9]: That's gonna be difficult tbh :-) let me check out the PR and have a brief look, then I'll approve.

File: src/webapp/package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (pabera) [reply src/webapp/package.json:31]: I think v17 doesn't bring us further. The LTS for v16 just has come to an end, the recommended version ist now v20. Have you tried this? Or at least the previous LTS v18?

This is independenz from the flag you have added here. I would have to double down on it in order to find a resolution.

[the remark this answers is not in the document]
Comment (pabera) [reply src/webapp/package.json:31]: I just noticed that you are not actually changing the Node version with your commit. My comment remains valid though it's irrelevant for the PR :-)

[the remark this answers is not in the document]
Comment (DivineDominion) [reply src/webapp/package.json:31]: I've used node v21.1 for the process. That's the default that was installed, btw, when grabbing future3/develop (or my branch) and building from scratch.

Can bump node to 20, too, if you like.

(I'm not even sure how far back it's technically possible to go, maybe on old hardware or OS's.)

**Comment by DivineDominion** [comment]
@pabera  Of course on another fresh install, things don't go as smooth today!

```
node:internal/modules/cjs/loader:452
      throw err;
      ^

Error: Cannot find module '/home/pi/RPi-Jukebox-RFID/src/webapp/node_modules/hoopy/index.js'. Please verify that the package.json has a valid "main" entry
    at tryPackage (node:internal/modules/cjs/loader:444:19)
    at Module._findPath (node:internal/modules/cjs/loader:715:18)
    at Module._resolveFilename (node:internal/modules/cjs/loader:1130:27)
    at Module._load (node:internal/modules/cjs/loader:985:27)
    at Module.require (node:internal/modules/cjs/loader:1235:19)
    at require (node:internal/modules/helpers:176:18)
    at Object.<anonymous> (/home/pi/RPi-Jukebox-RFID/src/webapp/node_modules/bfj/src/match.js:6:15)
    at Module._compile (node:internal/modules/cjs/loader:1376:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)
    at Module.load (node:internal/modules/cjs/loader:1207:32) {
  code: 'MODULE_NOT_FOUND',
  path: '/home/pi/RPi-Jukebox-RFID/src/webapp/node_modules/hoopy/package.json',
  requestPath: 'hoopy'
}

Node.js v21.1.0
```

So no need to test anything on your end just yet. I'll play around with this until it's fixed a.t.m.

**Comment by pabera** [comment]
No worries.

**Comment by DivineDominion** [comment]
The error was a fluke -- a user problem :)  I didn't notice that the `package-lock.json` was locally overwritten with the old values. Resetting that file to this branch/PR's content worked. Tried with a couple of `rm -rf node_modules` and `npm install` and `./run_rebuild.sh`. So it's good to go IMHO

**Comment by pabera** [comment]
I had a look at this PR. There are a few major version updates which can be risky. Specifically i18next v23 is a breaking version update and we can't guarantee that everything works smoothly. 

If you don't mind, I would like to stick to the same major versions for now. I have a new branch/PR open where I am updating dependencies for all aspects (node 20, python venv, bookworm)

https://github.com/MiczFlor/RPi-Jukebox-RFID/pull/2103

**Comment by DivineDominion** [comment]
Alright, thanks for looking into this!

**Pull request closed without merging**